src/huggingface_hub/hf_api.py

```diff
@@ -1737,6 +1737,9 @@ def upload_file(
         repo_type: Optional[str] = None,
         revision: Optional[str] = None,
         identical_ok: bool = True,
+        create_pr: Optional[bool] = False,
+        pr_commit_summary: Optional[str] = "",
+        pr_commit_description: Optional[str] = "",
```

Comment:
I'm not sure if creating a PR is in the scope of a function uploading a file. `create_pr` should be a separate method which creates a PR from existing files/branches I think.